enhancements/microshift/microshift-supportability.md

```diff
@@ -0,0 +1,384 @@
+
+## Motivation
+Most readers will be familiar with `must-gather` for OpenShift clusters and
+what it achieves. For MicroShift we need a tool with the same principles and
+goals, which is to gather every bit of information from the cluster to be able
+to diagnose and fix issues.
+
+### User Stories
+
+* As a MicroShift administrator, I want to be able to get a report about current
+  cluster status.
+* As a MicroShift developer, I want to be able to diagnose issues with a detailed
+  debug report.
+
+### Goals
+* Provide a single tool to automate the report creation.
+
+### Non-Goals
+N/A
+
+## Proposal
+The MicroShift gathering tool is a plugin in `sos`. This tool is delivered
+with RHEL, and is capable of generating reports for many applications in the
+system.
+
+Executing `sos report` yields information about a system and its components by
+using plugins to read each one of them. Plugins have an enablement check or may
+be invoked manually when executing the command.
+A [plugin](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin) is a
+python script that we include as part of `sos`.
+
+There is a [plugin for OpenShift](https://github.com/sosreport/sos/blob/main/sos/report/plugins/openshift.py)
+already that almost fits the needs of this EP. MicroShift does not share the
+same resources, APIs and namespaces/names that the plugin assumes, as it was
+made for OpenShift clusters. However, much of the needs covered in the plugin
+are the same, so the contents of the plugin may be used too.
+
+Apart from other components, that will be handled by their own plugins (full
+list in a system can be obtained by running `sos report -l`), we need to
+retrieve:
+1. Control plane component logs: apiserver, etcd, kubelet, etc.
+1. Yaml output for all the manifests.
+1. Logs for every container.
+1. MicroShift's configuration and version.
+
+Retrieving all this information involves using a kubernetes client. While
+MicroShift itself uses the go client extensively, the idea of introducing
+another command to produce this output and avoid installing `oc`/`kubectl` is
+not the best, since we would be adding runtime dependencies between a system
+component and MicroShift.
+
+### Workflow Description
+
+> _Disclaimer:_ Workflows described here illustrate manual processes.
+
+Assuming from now on MicroShift has already been installed in a R4E/RHEL based
+system and there has been some kind of issue.
+
+1. MicroShift administrator notices an issue/wants to get a system report.
+1. The administrator will log in to the failing host.
+1. The administrator will perform the debug report creation procedure:
+```
+sudo sos report
+```
+
+For the CI, the same procedure applies.
+
+#### Variation [optional]
+N/A
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+### Risks and Mitigations
+There are three main risks:
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. Extra required disk space.
```

Comment:
Should we note "not exposing user sensitive info" as a risk? The `sos` tool has `--clean` option, but I'm not sure how foolproof it is.

Reply:
The `--clean` option obfuscates usernames, hostnames, IPs, MAC addresses and keywords a usre passes as options. If we also restrict the amount of manifests we retrieve to only those that matter to diagnose platform issues we should be safe, right? (For example, avoid secrets)

Reply:
The details from this thread should be in the body of the doc.

Reply:
Added